### ISSUE TITLE
When user increments day of month past the end of the month, month should increment

### ISSUE THREAD

**Issue by jikamens** [jikamens/send-later#12]
In the date picker, if the user increments the day of the month past the end of the month, then the month should increment. E.g., currently displayed date is October 31, user increments day to 1, month should change to November. And, of course, if the previous month was December than the year should increment as well.

**Comment by jikamens** [comment]
Reported to Mozilla: https://bugzilla.mozilla.org/show_bug.cgi?id=1450238

**Comment by pya** [comment]
As a workaround or even a feature, would it be possible to display a message window if the send date is earlier than the current date? After all it is called "send later". So "send earlier" does not make much sense, as I can just send immediately without any add-on .  The cancel" button would bring me back to main dialog. There are  other options for repeated sending after a provided schedule. Maybe here it can make sense to have an earlier send date and the dialog is not appropriate. However, I have never used any repeated send option and would be happy to be reminded if I chose a send date in the past.

**Comment by jikamens** [comment]
>As a workaround or even a feature, would it be possible to display a message window if the send date is earlier than the current date?

There is already a [preference for this](https://blog.kamens.us/send-later/#prefs).

**Comment by pya** [comment]
Yes, setting *Don’t deliver messages more than # minutes late* does the trick. It took me while to figure this out. Of, course reading the documentation really helps. ;)  Maybe a candidate for a FAQ:

Q: How do I prevent sending a message (accidentally) scheduled for a past date?

A: Setting *Don’t deliver messages more than # minutes late* in the preferences shows you a warning message about this mail instead of sending it immediately.
